python/src/dataset.rs

```diff
@@ -581,13 +581,14 @@ impl Dataset {
             }
         };
 
+        let mut new_self = self.ds.as_ref().clone();
```

Comment:
Nit: maybe name it `ds` instead of `new_self` since it isn't really the same obj as self?  Also, isn't the `clone` redundant?  I think `self.ds.as_ref()` should give you `&mut LanceDataset`?

Reply:
`self.ds` is `Arc<LanceDataset`, which you can't get a `&mut` to. So we get immutable reference (`Arc<T>.as_ref()` returns `&T`) and then clone that to get an owned copy.